tinytodo/src/policy_store.rs

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2022-2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use std::{
+    path::{Path, PathBuf},
+    time::Duration,
+};
+
+use notify::{
+    event::{AccessKind, AccessMode},
+    Config, Error, Event, EventKind, RecommendedWatcher, Watcher,
+};
+use tokio::sync::mpsc::Sender;
+use tracing::{error, trace};
+
+use crate::context::{AppQuery, AppQueryKind};
+
+#[derive(Debug)]
+pub struct PolicySetWatcher {
+    watcher: RecommendedWatcher,
+    path: PathBuf,
+}
+
+impl PolicySetWatcher {
+    pub fn path(&'_ self) -> impl AsRef<Path> + '_ {
+        &self.path
+    }
+
+    #[tracing::instrument]
+    pub fn new(tx: Sender<AppQuery>, path: &Path) -> Self {
+        let config = Config::default().with_poll_interval(Duration::from_secs(1));
+        let watcher = RecommendedWatcher::new(
+            move |res: Result<Event, Error>| match res {
+                Ok(event) => {
+                    trace!("Event: {:?}", event);
```

Comment:
I don't think I want to see this event.

Reply:
You only see if you have the log level set to `trace`, one of the higher log levels. (Ie: setting it to `info` won't show this)

Reply:
And yes, `watcher` abstracts over the all the file watching APIs across different operating systems